solver/src/main/java/org/chocosolver/solver/variables/view/integer/IntAffineView.java

```diff
@@ -0,0 +1,588 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2023, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.variables.view.integer;
+
+import org.chocosolver.solver.ICause;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.delta.IIntDeltaMonitor;
+import org.chocosolver.solver.variables.delta.NoDelta;
+import org.chocosolver.solver.variables.events.IEventType;
+import org.chocosolver.solver.variables.events.IntEventType;
+import org.chocosolver.solver.variables.impl.scheduler.IntEvtScheduler;
+import org.chocosolver.solver.variables.view.IntView;
+import org.chocosolver.solver.variables.view.ViewDeltaMonitor;
+import org.chocosolver.util.iterators.DisposableRangeIterator;
+import org.chocosolver.util.iterators.DisposableValueIterator;
+import org.chocosolver.util.iterators.EvtScheduler;
+
+import static org.chocosolver.solver.variables.events.IntEventType.DECUPP;
+import static org.chocosolver.solver.variables.events.IntEventType.INCLOW;
+
+/**
+ * <br/>
+ *
+ * @author Charles Prud'homme
+ * @since 05/10/2023
+ */
+public final class IntAffineView<I extends IntVar> extends IntView<I> {
+
+    final boolean p; // positive
+    final int a;
+    final int b;
+
+
+    public static IntAffineView<IntVar> make(IntVar var, int a, int b) {
+        return new IntAffineView<>(var, a >= 0, Math.abs(a), b);
+    }
+
+    public static IntAffineView<IntVar> make(IntAffineView<IntVar> var, int a, int b) {
+        return new IntAffineView<>(var.getVariable(), var.p & (a >= 0), var.a * Math.abs(a), var.a * b + var.b);
+    }
+
+    /**
+     * <i>y</i> is an affine view of <i>x</i>: <i>y = (-1)a*x + b</i>.
+     *
+     * @param var a integer variable
+     */
+    private IntAffineView(final I var, boolean p, int a, int b) {
+        super((p ? "" : "-") + a + ".(" + var.getName() + ") + " + b, var);
+        assert a > 0;
+        this.p = p;
+        this.a = a;
+        this.b = b;
+    }
+
+    @Override
+    public IIntDeltaMonitor monitorDelta(ICause propagator) {
+        var.createDelta();
+        if (var.getDelta() == NoDelta.singleton) {
+            return IIntDeltaMonitor.Default.NONE;
+        }
+        return new ViewDeltaMonitor(var.monitorDelta(propagator)) {
+            @Override
+            protected int transform(int value) {
+                return (p ? value : -value) * a + b;
+            }
+        };
+    }
+
+
+    @Override
+    public boolean removeValue(int value, ICause cause) throws ContradictionException {
+        assert cause != null;
+        int inf = getLB();
+        int sup = getUB();
+        if (inf > value || value > sup) return false;
+        model.getSolver().getEventObserver().removeValue(this, value, cause);
```

Comment:
This is required for explanation purpose only, the explanation of var's modification is because the view has been modified in the first place, so the implication graph must reflect this order.